### PR TITLE
Optimize column mapping mode tests

### PR DESCRIPTION
**Pull request by pajaks**
In order to reduce execution time instead of execute all possible combinations some scenarios are selected.

## Description

Relates to #18805

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Comment by ebyhr** [comment]
Pushed an empty commit since `build-pt` failed. https://github.com/trinodb/trino/actions/runs/6035850480/job/16377032749?pr=18882

**Comment by ebyhr** [comment]
Just rebased on master to resolve conflicts.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java [hunk 1]
```diff
@@ -928,31 +928,31 @@ public Object[][] changeColumnMappingDataProvider()
...
                 ")"));
     }
 
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS}, dataProvider = "columnMappingDataProvider")
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS})
     @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
-    public void testDeltaSpecialCharacterColumnNamesWithColumnMappingMode(String mode)
+    public void testDeltaSpecialCharacterColumnNamesWithColumnMappingModeId()
     {
         testSpecialCharacterColumnNamesWithColumnMappingMode(tableName -> onDelta().executeQuery("" +
```
[the remark this answers is not in the document]
Comment (pajaks) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java:815]: I would do this in separate PR if thats ok.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java [hunk 2]
```diff
@@ -243,9 +243,9 @@ public void testColumnMappingModeNameWithNonLowerCaseColumn(String mode)
         }
     }
 
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS}, dataProvider = "columnMappingDataProvider")
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS})
     @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
-    public void testCreateTableWithNotNullColumn(String mode)
+    public void testColumnMappingModeIdWithNotNullColumn()
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java:248]: I would update `TestDeltaLakeConnectorTest#testTableWithNonNullableColumns` with dataProvider and remove this test.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java [hunk 3]
```diff
@@ -269,17 +269,17 @@ public void testCreateTableWithNotNullColumn(String mode)
...
+    public void testColumnMappingModeIdCreatePartitionTableAsSelect()
     {
         String tableName = "test_dl_create_partition_table_as_select_" + randomNameSuffix();
 
         onTrino().executeQuery("" +
                 "CREATE TABLE delta.default." + tableName + " " +
                 "WITH (" +
                 "location = 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'," +
-                "column_mapping_mode = '" + mode + "'" +
+                "column_mapping_mode = 'id'" +
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java:282]: The name column mapping mode is more important than id in Delta/Databricks. The id mapping is basically for compatibility with Iceberg. I feel running only with id mapping doesn't make sense.

**Comment by ebyhr** [comment]
/test-with-secrets sha=6c76874175ec8a41a6efdcd1fce87293f26069b3

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java [hunk 4]
```diff
@@ -410,6 +410,21 @@ public void testDropAndAddColumnWithSameName()
                 .hasMessageContaining("Cannot drop column from table using column mapping mode NONE");
     }
 
+    @Test(dataProvider = "columnMappingModeDataProvider")
+    public void testDropPartitionColumn(ColumnMappingMode mode)
```
Comment (findinpath) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:414]: One small ask:
In case of the product tests which are being replaced with integration tests, please do this in a separate commit to distinguish easier what the changes in TestDeltaLakeColumnMappingMode.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java [hunk 5]
```diff
@@ -1151,6 +1166,80 @@ private void testCreateTableColumnMappingMode(ColumnMappingMode mode, Consumer<S
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test(dataProvider = "columnMappingModeDataProvider")
+    public void testDropAndAddColumnShowStatsForColumnMappingMode(ColumnMappingMode mode)
+    {
+        if (mode == ColumnMappingMode.NONE) {
+            throw new SkipException("The connector doesn't support dropping columns with 'none' column mapping");
```
Comment (findinpath) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:1173]: nit: It is actually the Delta table format which doesn't support this operation.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java [hunk 6]
```diff
@@ -1020,60 +937,6 @@ private void assertDeltaTrinoTableEquals(String tableName, String trinoQuery, St
...
-                    .hasMessageMatching("(?s).*(Cannot drop partition column: part|Dropping partition columns \\(part\\) is not allowed).*");
-            assertThat(getTablePropertyOnDelta("default", tableName, "delta.columnMapping.maxColumnId"))
-                    .isEqualTo("3");
-
-            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + tableName))
-                    .containsOnly(row(1, "part#1"));
-            assertThat(onDelta().executeQuery("SELECT * FROM default." + tableName))
-                    .containsOnly(row(1, "part#1"));
-
-            // Verify adding a new column with the same name doesn't allow accessing the old data
```
Comment (findinpath) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java:1141]: We have a slightly similar coverage for dropping and adding a new column with the same name in `io.trino.plugin.deltalake.TestDeltaLakeConnectorTest#testDropAndAddColumnShowStatsForColumnMappingMode`.

Please add though also an integration test which verifies this specific use case.

**Review by findinpath — COMMENTED** [review]
LGTM % comments

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java [hunk 7]
```diff
@@ -1151,6 +1166,90 @@ private void testCreateTableColumnMappingMode(ColumnMappingMode mode, Consumer<S
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test(dataProvider = "columnMappingModeDataProvider")
+    public void testDropAndAddColumnShowStatsForColumnMappingMode(ColumnMappingMode mode)
+    {
+        if (mode == ColumnMappingMode.NONE) {
+            throw new SkipException("DeltaLake connector doesn't support dropping columns with 'none' column mapping");
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:1173]:
```suggestion
            throw new SkipException("Delta Lake doesn't support dropping columns with 'none' column mapping");
```

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java [hunk 8]
```diff
@@ -292,9 +266,9 @@ public void testCreatePartitionTableAsSelect(String mode)
         }
     }
 
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS}, dataProvider = "columnMappingDataProvider")
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS})
     @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
-    public void testCreateTableWithComments(String mode)
+    public void testCreateTableWithCommentsColumnMappingModeName()
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java:271]: Do we have an integration test with column mapping mode id?

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java [hunk 9]
```diff
@@ -318,38 +292,20 @@ public void testCreateTableWithComments(String mode)
         }
     }
 
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS}, dataProvider = "columnMappingDataProvider")
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS})
     @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
-    public void testTrinoColumnMappingModeCommentOnTable(String mode)
+    public void testColumnMappingModeNameCommentOnTable()
```
[the remark this answers is not in the document]
Comment (pajaks) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java:297]: Added

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java [hunk 10]
```diff
@@ -1151,6 +1166,128 @@ private void testCreateTableColumnMappingMode(ColumnMappingMode mode, Consumer<S
...
+
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, 10), (2, 20), (null, null)", 3);
+        assertUpdate("ANALYZE " + tableName);
+
+        assertQuery("SHOW STATS FOR " + tableName, "VALUES" +
+                "('a_number', null, 2.0, 0.33333333333, null, 1, 2)," +
+                "('b_number', null, 2.0, 0.33333333333, null, 10, 20)," +
+                "(null, null, null, null, 3.0, null, null)");
+
+        // Ensure SHOW STATS return the same stats for the restored column
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:1236]: restored → renamed

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java [hunk 11]
```diff
@@ -1151,6 +1166,128 @@ private void testCreateTableColumnMappingMode(ColumnMappingMode mode, Consumer<S
...
+    }
+
+    @Test(dataProvider = "columnMappingModeDataProvider")
+    public void testCommentOnTableForColumnMappingMode(ColumnMappingMode mode)
+    {
+        String tableName = "test_comment_on_table_for_column_mapping_mode_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (a_number INT, b_number INT) WITH (column_mapping_mode='" + mode + "')");
+
+        assertUpdate("COMMENT ON TABLE " + tableName + " IS 'test comment' ");
+        assertQuery("SELECT comment FROM system.metadata.table_comments WHERE table_name = '" + tableName + "'", "VALUES ('test comment')");
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:1260]: There's a `getTableComment` method.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java [hunk 12]
```diff
@@ -1151,6 +1166,128 @@ private void testCreateTableColumnMappingMode(ColumnMappingMode mode, Consumer<S
...
+    }
+
+    @Test(dataProvider = "columnMappingModeDataProvider")
+    public void testCommentOnColumnForColumnMappingMode(ColumnMappingMode mode)
+    {
+        String tableName = "test_comment_on_column_for_column_mapping_mode_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (a_number INT, b_number INT) WITH (column_mapping_mode='" + mode + "')");
+
+        assertUpdate("COMMENT ON COLUMN " + tableName + ".a_number IS 'test column comment'");
+        assertQuery("SELECT comment FROM delta.information_schema.columns WHERE table_name = '" + tableName + "' AND column_name = 'a_number'", "VALUES ('test column comment')");
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:1272]: There's a `getColumnComment` method.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java [hunk 13]
```diff
@@ -1151,6 +1166,128 @@ private void testCreateTableColumnMappingMode(ColumnMappingMode mode, Consumer<S
...
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test(dataProvider = "columnMappingModeDataProvider")
+    public void testCreateTableWithCommentsColumnForColumnMappingMode(ColumnMappingMode mode)
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:1278]: nit: I would rename to `testCreateTableWithCommentsForColumnMappingMode` (remove Column) because this test contains a table comment.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java [hunk 14]
```diff
@@ -1151,6 +1166,128 @@ private void testCreateTableColumnMappingMode(ColumnMappingMode mode, Consumer<S
...
+    {
+        String tableName = "test_create_table_with_comments_for_column_mapping_mode_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (a_number INT COMMENT 'test column comment', b_number INT)  " +
+                "COMMENT 'test table comment' " +
+                "WITH (column_mapping_mode='" + mode + "')");
+
+        assertQuery("SELECT comment FROM system.metadata.table_comments WHERE table_name = '" + tableName + "'", "VALUES ('test table comment')");
+        assertQuery("SELECT comment FROM delta.information_schema.columns WHERE table_name = '" + tableName + "' AND column_name = 'a_number'", "VALUES ('test column comment')");
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:1286]: There're `getTableComment` & `getColumnComment` methods.

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
/test-with-secrets sha=f645bed402d07c979cb4fd8dec31220bf9b5f4f4
